src/periodic.jl

```diff
@@ -12,18 +15,21 @@ Applies a periodic function `f` with provided `period` and `phase_shift` to the
 # Fields
 * `f::Union{typeof(cos), typeof(sin)}`: the periodic function
 * `period::Union{Real, Period}`: the function period. Must be strictly positive.
-* `phase_shift::Union{Real, Period}`: adjusts the phase of the periodic function, measured
-    in the same units as the input. Increasing the value translates the function to the
-    right, toward higher/later input values.
+* `phase_shift::Union{Real, Period}` (optional): adjusts the phase of the periodic function,
+  measured in the same units as the input. Increasing the value translates the function to
+  the right, toward higher/later input values.
 """
-struct Periodic{T} <: Transform where T <: Union{Real, Period}
+struct Periodic{P, S} <: Transform
     f::Union{typeof(cos), typeof(sin)}
-    period::T
-    phase_shift::T
+    period::P
+    phase_shift::S
 
-    function Periodic(f, period::T, phase_shift::T) where T
-        period > zero(T) || throw(ArgumentError("period must be strictly positive."))
-        return new{T}(f, period, phase_shift)
+    function Periodic(f, period::P, phase_shift::S) where {P, S}
+        if !((P <: Real && S <: Real) || (P <: Period && S <: Period))
+            throw(ArgumentError("period and phase_shift must have the same supertype"))
```

Comment:
I guess this gives a more informative error message compared to my suggestion which would have just been a MethodError.